### PR TITLE
feat(#10): add sharepoint data import

### LINKED ISSUE #10: Add import for manual measurements
We can access Excel Worksheets via the [Microsoft Graph API](https://learn.microsoft.com/en-us/graph/use-the-api?context=graph%2Fapi%2F1.0&view=graph-rest-1.0):

1. fetch a token using a registered azure application
2. use the link of a shared excel document
3. read the values from the first worksheet of the shared document and returns it

To test, call:

```bash
curl http://localhost:3000/graph/samples | jq "."
```

Make sure to configure this in your `.env`:

```env
GRAPH_TENANT_ID=<tenant-id>
GRAPH_CLIENT_ID=<client-id>
GRAPH_CLIENT_SECRET=<client-secret>
GRAPH_ITEM_SHARE_URL=<document-share-link>
```

The result it a json like this:

```json
[
  [
    "Datum",
    "Uhrzeit",
    "Lat",
    "Lng",
    "Alt",
    "Probennummer",
    "Stoffname",
    "Messwert"
  ],
  [
    45624,
    0.5625,
    8.2135422,
    53.1446486,
    7,
    1,
    "Stickstoff",
    0.3
  ],
  [
    45624,
    0.5625,
    8.2135422,
    53.1446486,
    7,
    2,
    "Phosphor",
    0.07
  ]
]
```

## Performance improvements

This process involves up to 4 http calls, which may entail wait times in the order of 6 seconds. That's why it may be beneficial to fetch the data and store it in memory/on disk.

> ⚠️ Updates might be possible to achieve using a webhooks API: https://learn.microsoft.com/en-us/graph/change-notifications-delivery-webhooks

### PR DESCRIPTION
**Pull request by jhohlfeld**
closes #10

### CODE REVIEW

**Review by oaris-dev — APPROVED** [review]

File: package.json
Comment (jhohlfeld) [inline package.json:1]: Add test script

**Comment by jhohlfeld** [comment]
We need webhook because: slow! (6s)
https://learn.microsoft.com/en-us/graph/change-notifications-delivery-webhooks?tabs=http

File: app/graph/samples/route.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import { fetchWorksheetData } from '@/app/actions'
+import { graphItemShareUrl } from '@/config'
+import { hasGraphErrorCause } from '@/util/error'
+
+/**
+ * Router handler for receiving token responses from MS Azure application auth.
+ */
```
Comment (jhohlfeld) [inline app/graph/samples/route.ts:7]:
```suggestion
```

**Comment by jhohlfeld** [comment]
@oaris-dev I'll close this issue now and will add the subscription/webhook in another issue

**Pull request merged**